controllers/volumereplicationgroup_controller.go

```diff
@@ -1145,6 +1173,103 @@ func (v *VRGInstance) createVR(vrNamespacedName types.NamespacedName, state volr
 	return nil
 }
 
+// namespacedName applies to both VolumeReplication resource and pvc as of now.
+// This is because, VolumeReplication resource for a pvc that is created by the
+// VolumeReplicationGroup has the same name as pvc. But in future if it changes
+// functions to be changed would be processVRAsPrimary(), processVRAsSecondary()
+// to either receive pvc NamespacedName or pvc itself as an additional argument.
+func (v *VRGInstance) selectVolumeReplicationClass(namespacedName types.NamespacedName) (string, error) {
+	className := ""
+
+	if !v.vrcUpdated {
```

Comment:
(nit) as replClassList is a pointer a `nil` check instead of a bool would also work here? Not a deal breaker, just wondering.

Reply:
I guess the reason why instead of a nil pointer check for replClassList and a separate bool variable to ensure that we dont do repeated updateReplicationClassList when there are no replication classes present. This boolean helps in determining that listing VolumeReplicationClasses was already done in this reconcile loop and need not be done again. If the nil check is made we cant distinguish between updateReplicationClassList() being not called and no replication classes being present.